src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -467,6 +465,172 @@ class CVulkanLogicalDevice final : public ILogicalDevice
         return !anyFailed;
     }
 
+    void updateDescriptorSets_impl(uint32_t descriptorWriteCount, const IGPUDescriptorSet::SWriteDescriptorSet* pDescriptorWrites, uint32_t descriptorCopyCount, const IGPUDescriptorSet::SCopyDescriptorSet* pDescriptorCopies)
+    {
+        constexpr uint32_t MAX_DESCRIPTOR_ARRAY_COUNT = 256u;
+
+        core::vector<VkWriteDescriptorSet> vk_writeDescriptorSets(descriptorWriteCount);
+
+        uint32_t bufferInfoOffset = 0u;
+        core::vector<VkDescriptorBufferInfo >vk_bufferInfos(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
+
+        uint32_t imageInfoOffset = 0u;
+        core::vector<VkDescriptorImageInfo> vk_imageInfos(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
+
+        uint32_t bufferViewOffset = 0u;
+        core::vector<VkBufferView> vk_bufferViews(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
+
+        core::vector<VkWriteDescriptorSetAccelerationStructureKHR> vk_writeDescriptorSetAS(descriptorWriteCount);
+
+        uint32_t accelerationStructuresOffset = 0u;
+        core::vector<VkAccelerationStructureKHR> vk_accelerationStructures(descriptorWriteCount * MAX_DESCRIPTOR_ARRAY_COUNT);
```

Comment:
same logic as here https://github.com/Devsh-Graphics-Programming/Nabla/pull/345/files#r1047770830 applies to everything, no offsets needed

Reply:
TODO

Reply:
Resolved.